phpBB/phpbb/search/backend/search_backend_interface.php

```diff
@@ -0,0 +1,155 @@
+	* @param	array		$sort_by_sql		contains SQL code for the ORDER BY part of a query
+	* @param	string		$sort_key			is the key of $sort_by_sql for the selected sorting
+	* @param	string		$sort_dir			is either a or d representing ASC and DESC
+	* @param	string		$sort_days			specifies the maximum amount of days a post may be old
+	* @param	array		$ex_fid_ary			specifies an array of forum ids which should not be searched
+	* @param	string		$post_visibility	specifies which types of posts the user can view in which forums
+	* @param	int			$topic_id			is set to 0 or a topic id, if it is not 0 then only posts in this topic should be searched
+	* @param	array		$author_ary			an array of author ids
+	* @param	string		$author_name		specifies the author match, when ANONYMOUS is also a search-match
+	* @param	array		&$id_ary			passed by reference, to be filled with ids for the page specified by $start and $per_page, should be ordered
+	* @param	int			$start				indicates the first index of the page
+	* @param	int			$per_page			number of ids each page is supposed to contain
+	* @return	boolean|int						total number of results
+	*/
+	public function author_search($type, $firstpost_only, $sort_by_sql, $sort_key, $sort_dir, $sort_days, $ex_fid_ary, $post_visibility, $topic_id, $author_ary, $author_name, &$id_ary, &$start, $per_page);
+
+	/**
+	* Updates wordlist and wordmatch tables when a message is posted or changed
+	*
+	* @param	string		$mode contains the post mode: edit, post, reply, quote ...
+	* @param	int			$post_id	contains the post id of the post to index
+	* @param	string		$message	contains the post text of the post
+	* @param	string		$subject	contains the subject of the post to index
+	* @param	int			$poster_id	contains the user id of the poster
+	* @param	int			$forum_id	contains the forum id of parent forum of the post
+	*/
+	public function index($mode, $post_id, &$message, &$subject, $poster_id, $forum_id);
+
+	/**
+	* Destroy cached results, that might be outdated after deleting a post
+	*/
+	public function index_remove($post_ids, $author_ids, $forum_ids);
+
+	/**
+	* Destroy old cache entries
+	*/
+	public function tidy();
+
+	/**
+	* Create fulltext index
+	*
+	* @return string|bool error string is returned incase of errors otherwise false
+	*/
+	public function create_index($acp_module, $u_action);
+
+	/**
+	* Drop fulltext index
+	*
+	* @return string|bool error string is returned incase of errors otherwise false
+	*/
+	public function delete_index($acp_module, $u_action);
+
+	/**
+	* Returns true if both FULLTEXT indexes exist
+	*/
+	public function index_created();
+
+	/**
+	* Returns an associative array containing information about the indexes
+	*
+	* @return array|bool Language string of error false otherwise
+	*/
+	public function index_stats();
+
+	/**
+	* Display various options that can be configured for the backend from the acp
+	*
+	* @return associative array containing template and config variables
```

Comment:
Forgot to specify the actual type

Reply:
That will be removed from there probably